segmentation/utils/segmentation_utils_test.py

```diff
@@ -182,6 +182,136 @@ def test_watershed_transform():
+                                                                  transform_kwargs=transform_kwargs)
+
+    assert np.array_equal(arcsinh_data.loc[:, :, unchanged_cols].values,
+                          cell_data.loc[:, :, unchanged_cols].values)
+
+    # TODO: In general it's bad practice for tests to call the same function as code under test
+    for cell in cell_data.cell_id:
+        arcsinh_vals = np.arcsinh(cell_data.loc[:, cell, modified_cols].values)
+        assert np.array_equal(arcsinh_data.loc[:, cell, modified_cols].values, arcsinh_vals)
+
+
+def test_transform_expression_matrix_multiple_compartments():
+    # create expression matrix
+    cell_data = np.random.choice([0, 1, 2, 3, 4], 140, replace=True)
+    cell_data = cell_data.reshape((2, 10, 7)).astype('float')
+
+    coords = [['whole_cell', 'nuclear'], list(range(10)),
+              ['cell_size', 'chan1', 'chan2', 'chan3', 'label', 'morph_1', 'morph_2']]
+    dims = ['compartments', 'cell_id', 'features']
+
+    cell_data = xr.DataArray(cell_data, coords=coords, dims=dims)
+
+    unchanged_cols = ['cell_size', 'label', 'morph_1', 'morph_2']
+    modified_cols = ['chan1', 'chan2', 'chan3']
+
+    # test size_norm
+    normalized_data = segmentation_utils.transform_expression_matrix(cell_data,
+                                                                     transform='size_norm')
+
+    assert np.array_equal(normalized_data.loc[:, :, unchanged_cols].values,
+                          cell_data.loc[:, :, unchanged_cols].values)
+
+    # TODO: In general it's bad practice for tests to call the same function as code under test
+    for cell in cell_data.cell_id:
+        if cell_data.loc['whole_cell', cell, 'cell_size'] != 0:
+            normalized_vals = np.divide(cell_data.loc['whole_cell', cell, modified_cols].values,
+                                        cell_data.loc['whole_cell', cell, 'cell_size'].values)
+            assert np.array_equal(normalized_data.loc['whole_cell', cell, modified_cols].values,
+                                  normalized_vals)
+
+    # test arcsinh transform
+    transform_kwargs = {'linear_factor': 1}
+    arcsinh_data = segmentation_utils.transform_expression_matrix(cell_data,
+                                                                  transform='arcsinh',
+                                                                  transform_kwargs=transform_kwargs)
+
+    assert np.array_equal(arcsinh_data.loc[:, :, unchanged_cols].values,
+                          cell_data.loc[:, :, unchanged_cols].values)
+
+    # TODO: In general it's bad practice for tests to call the same function as code under test
+    for cell in cell_data.cell_id:
+        arcsinh_vals = np.arcsinh(cell_data.loc[:, cell, modified_cols].values)
+        assert np.array_equal(arcsinh_data.loc[:, cell, modified_cols].values, arcsinh_vals)
+
+
+# TODO: The testing for this function can be improved. Repeated code, and lots of manual checks
```

Comment:
to avoid repeated code here, the setup/initialization could be put into an internal helper function, with a flag for how many compartments to add.  

As for the 'same function as code' bad practice, I don't see a reasonable way around it here.  Only thing that comes to mind is that if a set seed is used for the randomized input, we can pre-compute and store the desired output in a file to load during testing, but this seems like overkill for making sure that `np.arcsinh` still works as expected.

Reply:
Agreed. I think maybe switching to doing it line-by-line rather than the whole thing at once is enough of a difference? Just want to catch any dumb mistakes